entity-framework/efcore-and-ef6/porting/port-detailed-cases.md

```diff
@@ -0,0 +1,149 @@
+
+EF6 supported custom ("lightweight") conventions and model conventions. The lightweight conventions are similar to EF Core's [pre-convention model configuration](/ef/core/what-is-new/ef-core-6.0/whatsnew#pre-convention-model-configuration). Other conventions are supported as part of model building.
+
+EF6 runs conventions after the model is built. EF Core applies them as the model is being built. In EF Core, you can decouple model building from active sessions with a DbContext, but unlike EF6 you must pass your conventions to the model builder.
+
+## Data validation
+
+EF Core does not support data validation and only uses data annotations for building the model and migrations. Most client libraries from web/MVC to WinForms and WPF provide a data validation implementation to use.
+
+## Features that are coming soon
+
+There are a few features in EF6 that don't exist yet in EF Core, but are on the product roadmap.
+
+- **Table-per-concrete type (TPC)** was supported in EF6 along with "entity splitting." TPC is on the roadmap for EF7.
+- **Stored procedure mapping** in EF6 allows you to delegate create, update, and delete operations to stored procedures. EF Core currently only allows mapping to stored procedures for reads. Create, update, and delete (CUD) support is on the roadmap for EF7.
+- **Complex types** in EF6 are similar to owned types in EF Core. However, the full set of capabilities will be addressed with value objects in EF7.
+
+## Leave ObjectContext behind
+
+EF Core uses a [DbContext](/ef/core/dbcontext-configuration) instead of an `ObjectContext`. You will have to update code that uses [IObjectContextAdapter](xref:System.Data.Entity.Infrastructure.IObjectContextAdapter). This was sometimes used for queries with `PreserveChanges` or `OverwriteChanges` merge option. For similar capabilities in EF Core, look into the [Reload](xref:Microsoft.EntityFrameworkCore.ChangeTracking.EntityEntry.Reload) method.
+
+## Model configuration
+
+There are many important differences between how models in EF6 and EF Core are designed. EF Core lacks full support for conditional mapping. It does not have model builder versions. For spatial support, EF Core recommends using a third-party library such as [NetTopologySuite](https://github.com/NetTopologySuite/NetTopologySuite).
+
+Other differences include:
+
+### Type discovery
+
+In EF Core, Entity Types are discovered by the engine in three ways:
+
+- Expose a `DbSet<TEntity>` on your `DbContext` where `TEntity` is the type you wish to track.
+- Reference a `Set<TEntity>` from somewhere in your code.
+- Complex types referenced by discovered types are recursively discovered (for example, if your `Blog` references a `Post` and `Blog` is discoverable, `Post` will be discovered as well)
+
+### Mapping
+
+The `.Map()` extension in EF6 has been replaced with overloads and extension methods in EF Core. For example, to call a stored procedure you can use the `FromSqlRaw()` method on the `DbSet<>` instance.
```

Comment:
The example is not relevant.  `Map` is usually used to configure inheritance mapping, so as an example `HasDiscriminator` can be used to configure TPH